cartography/intel/github/repos.py

```diff
@@ -214,6 +227,62 @@ def _transform_collaborators(collaborators, repo_url, transformed_collaborators)
             transformed_collaborators[user_permission].append(user)
 
 
+def _transform_python_requirements(repo_object, out_requirements_files):
+    """
+    Performs data transformations for the requirements.txt files in a GitHub repo, if available.
+    :param repo_object: The repo object.
+    :param out_requirements_files: Output array to append transformed results to.
+    :return: Nothing.
+    """
+    req_file_contents = repo_object['requirements']
+    if req_file_contents and req_file_contents.get('text'):
+        text_contents = req_file_contents['text']
+
+        parsed_list = []
+        for line in text_contents.split("\n"):
+            try:
+                # Skip over empty lines
+                if line:
+                    req = Requirement(line)
+                    parsed_list.append(req)
+            except ValueError as e:
+                logger.info(
+                    f"Failed to parse line \"{line}\" in repo {repo_object['url']}'s requirements.txt; skipping. "
+                    f"Details: {e}.",
+                )
+                continue
+
+        for req in parsed_list:
+            if not req.name:
+                logger.info(
+                    f"Could not get library name for an object in {repo_object['url']}'s requirements.txt, skipping.",
+                )
+                continue
+
+            # Set `spec` to a default value. Example values for str(req.specifier): "<4.0,>=3.0" or "==1.0.0".
+            spec = str(req.specifier)
+
+            if len(req.specifier._specs) == 1:
+                # req.specifier._specs is a frozenset so to manipulate it we turn it into a list and get the 1st item.
+                spec_str = str(list(req.specifier._specs)[0])
+                # If the spec is pinned to 1 version with ==, we only want the version number, not the full specifier.
+                spec = spec_str.strip('==') if spec_str.startswith('==') else spec
+
+            # Ingest `None` to the graph instead of empty string.
+            if spec == '':
+                spec = None
+
+            canon_name = canonicalize_name(req.name)
+
+            out_requirements_files.append({
+                "id": canon_name,
+                "name": canon_name,
+                "specifier": spec,
+                "url": req.url,
```

Comment:
IMO we shouldn't support this (at least in V1) - we don't use this internally. (We do have some URLs in our requirements files but it's actually a slightly different format to this.)